File: Problem Sets/PSet 2/Readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>

int main(void)
{
    int letters = 0, words = 1, sent = 0;
    string text = get_string("Type your text: ");
	
// count
    for (int i = 0; i < strlen(text); i++)
    {   
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z')) // just letters uppercase or lowercase
        {
            letters++;
        }   
        else if (text[i] == ' ') // se tiver espaço, acrescente +1 na contagem de palavras
        {
            words++;
        }
        else if (text[i] == '!' || text[i] == '?' || text[i] == '.') // se tiver ! ? ou . acrescente +1 na contagem de frases 
        {
            sent++;
        }
    }
    
// Readability test - Coleman-Liau index | index = 0.0588 * L - 0.296 * S - 15.8
    float grade = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sent / (float) words) - 15.8;
// Printing
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 16 && grade >= 0)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: Problem Sets/PSet 6/Readability/readability.py
import re
from cs50 import get_string
from cs50 import get_int


def main():
    text = get_string("Type your text: ")
    charaAlpha = 0

# count with built-in method isalph()
    for count in text:
        if count.isalpha():
            charaAlpha += 1

    # Regex with function findall
    sent = len(re.findall(r'\!|\?|\.', text))

    # Method split
    words = len(text.split())

    # the Coleman-Liau formula 0.0588 * L - 0.296 * S - 15.8
    grade = round((0.0588 * (charaAlpha/words*100)) - (0.296 * (sent/words*100)) - 15.8)

    if (grade >= 16):
        print("Grade 16+\n")
    elif (grade >= 0 and grade <= 15):
        print("Grade: ", + grade)
    else:
        print("Before Grade 1")


if __name__ == "__main__":
    main()
